pallets/staking/src/lib.rs

```diff
@@ -443,43 +446,34 @@ pub mod pallet {
 					let created_at = Self::get_period_number(position.created_at)
 						.defensive_ok_or::<Error<T>>(InconsistentStateError::Arithmetic.into())?;
 
-					let (claimable_rewards, claimable_unpaid_rewards, unpaid_rewards, payable_percentage) =
-						Self::calculate_rewards(
-							position,
-							staking.accumulated_reward_per_stake,
-							current_period,
-							created_at,
-						)
-						.ok_or(Error::<T>::Arithmetic)?;
-
-					let rewards = claimable_rewards
-						.checked_add(claimable_unpaid_rewards)
-						.ok_or(Error::<T>::Arithmetic)?;
-
-					let pot = Self::pot_account_id();
-					T::Currency::transfer(T::NativeAssetId::get(), &pot, &who, rewards)?;
-
-					position.accumulated_unpaid_rewards = position
-						.accumulated_unpaid_rewards
-						.checked_add(unpaid_rewards)
-						.ok_or(Error::<T>::Arithmetic)?;
-					position.accumulated_unpaid_rewards = position
-						.accumulated_unpaid_rewards
-						.checked_sub(claimable_unpaid_rewards)
-						.defensive_ok_or::<Error<T>>(InconsistentStateError::NegativeUnpaidRewards.into())?;
-
-					position.accumulated_locked_rewards = position
-						.accumulated_locked_rewards
-						.checked_add(rewards)
-						.ok_or(Error::<T>::Arithmetic)?;
+					let (rewards, unpaid_rewards, payable_percentage) = Self::calculate_rewards(
+						position,
+						staking.accumulated_reward_per_stake,
+						current_period,
+						created_at,
+					)
+					.ok_or(Error::<T>::Arithmetic)?;
+
+					if !rewards.is_zero() {
+						let pot = Self::pot_account_id();
+						T::Currency::transfer(T::NativeAssetId::get(), &pot, &who, rewards)?;
+
+						position.accumulated_locked_rewards = position
+							.accumulated_locked_rewards
+							.checked_add(rewards)
+							.ok_or(Error::<T>::Arithmetic)?;
```

Comment:
is this correct? 

to me, it looks like you send the `rewards` to `who` and then you add the same amount to locked rewards ? 

Doesnt this mean that it has more accumulated rewards than he should ?

Reply:
This is correct. Increase stake locks rewards, so we send new rewards to user and then we locked them here: https://github.com/galacticcouncil/HydraDX-node/blob/0b9b0f1bba50d8e2cf4f6bb79c1aceb2c843d58b/pallets/staking/src/lib.rs#L496